specs/data-grid/data-grid.spec.md

```diff
@@ -0,0 +1,302 @@
+- none
+
+*parts:*
+- `cellsSlot`
+
+**Data grid cell**
+*Component name:*
+- `fast-data-grid-cell`
+
+*Attributes:*
+- `gridColumnIndex`  
+The grid column this cell appears in.
+
+*properties:*
+- `rowData`  
+The object that contains the data to be displayed in this row.
+
+- `columnData`  
+The `DataGridColumn` this cell represents.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `cellSlot`
+
+## Implementation
+
+The most basic use case for this component would be to just pass it an array of objects for it to render.  
+
+For example given a grid component:
+```html
+ <fast-data-grid id="defaultGrid"></fast-data-grid>
+ ```
+
+ And some data:
+ ```js
+const baseRows: object[] = [
+    { name: "Rob", age: "19" },
+    { name: "Bob", age: "20" },
+];
+ ```
+
+ An author could pass the data to the component from a javascript function:
+
+```js
+onLoad(): void {
+   const defaultGrid: DataGrid | null = document.getElementById(
+        "defaultGrid"
+    ) as DataGrid;
+    if (defaultGrid !== null) {
+        defaultGrid.rowsData = baseRows;
+    }
+}
+```
+
+This renders a basic grid with a column titled "name" and another titled "age" in addition to the two rows of data populated with the values.  
+
+![](./images/ex1.png)
+
+The next level of customization involves changing the default columns that are created by the component when none are provided.
+
+And author would define the columns by providing an array of `DataGridColumn` objects to the component's `columnsData` property:
+
+```js
+const baseColumns: DataGridColumn[] = [
+    { columnDataKey: "name", title:"Player name", columnWidth: "1fr" },
+    { columnDataKey: "age", title:"Age", columnWidth: "80px" },
+];
+
+...
+    defaultGrid.columnsData = baseColumns;
+...
+```
+
+Applying these columns to our previous example results in our columns having the new titles applied as well as having a fixed width for the "Player age" column.
+
+![](./images/ex2.png)
+
+- programmatically generated rows/cells will will be created using [repeat directives](https://fast.design/docs/fast-element/using-directives#the-repeat-directive)
+
+- individual cells can be customized using by passing a custom ViewTemplate through the `DataGridColumn` interface for the column in question. These templates are rendered in the light dom within the cell so that authors can create custom cells with interactive elements.
+
+### Accessibility
+
+The `data grid` should align to the interaction model provided by the [W3C](https://w3c.github.io/aria-practices/#grid).
+
+### Globalization
+
+The order of grid columns is inverted in RTL layouts.
+
+### Security
+
+NA
+
+### Performance
+
+We want to take full advantage of fast-element templating and directives for fast rendering.
+
+### Dependencies
+
+None
+
+### Test Plan
+
+While testing is still TBD for our web components, I would expect this to align with the testing strategy and not require any additional test support.
```

Comment:
Can you reword this sentence a little? The words "I" or any other authors "speaker voice" should not be present in a spec. I think we've been a little lax on this in the spec files but they should read fairly "voice" agnostic.

We also do have a testing strategy now for components as put in place by @chrisdholt so this could read: "This component should have Jest component testing in the `@microsoft/fast-foundation` package."

Reply:
Before you add the above, just note that we don't use *Jest* for web components ;)

Reply:
My bad, I guess it's just `Mocha`.